ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -372,12 +371,17 @@ public void onCompletion(String result, Exception exception) {
               router.deleteBlob(result, deleteCallback);
               break;
             default:
-              throw new IllegalStateException("Unrecognized RestMethod: " + restMethod);
+              exception = new IllegalStateException("Unrecognized RestMethod: " + restMethod);
           }
+        } catch (Exception e) {
+          exception = e;
         }
-      } finally {
-        callbackTracker.markCallbackProcessingEnd();
       }
+
+      if (exception != null) {
+        submitResponse(restRequest, restResponseChannel, null, exception);
```

Comment:
The intent of this change is to make sure all requests get responses. If were to throw from here, the caller of this function is the `IdConverter` that would just probably just swallow the exception. In that case, the request that suffered the exception would not get a response.

With this change, all requests will get responses even if there was an exception here.